README.md

```diff
@@ -1,6 +1,69 @@
 
-[![Nuts](https://circleci.com/gh/nuts-foundation/go-did.svg?style=svg)](https://circleci.com/gh/nuts-foundation/go-did)
-
-# go-did
+# github.com/nuts-foundation/go-did [![Nuts](https://circleci.com/gh/nuts-foundation/go-did.svg?style=svg)](https://circleci.com/gh/nuts-foundation/go-did) [![Go Reference](https://pkg.go.dev/badge/github.com/nuts-foundation/go-did.svg)](https://pkg.go.dev/github.com/nuts-foundation/go-did)
 
 A library to parse and generate W3C DID Documents.
+
+## Example usage:
+Creation of a simple DID Document which is its own controller and contains an AssertionMethod.
+```go
+didID, err := did.ParseDID("did:example:123")
+if err != nil {
+    panic(err)
+}
+
+doc := &did.Document{
+    Context:            []did.URI{did.DIDContextV1URI()},
+    ID:                 *didID,
+    Controller:         []did.DID{*didID},
+}
+
+keyID, err := did.ParseDID(didID.String() + "#key-1")
+if err != nil {
+    panic(err)
+}
+
+keyPair, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+if err != nil {
+    panic(err)
+}
+verificationMethod, err := did.NewVerificationMethod(*keyID, did.JsonWebKey2020, did.DID{}, keyPair.Public())
```

Comment:
If the third argument (controller?) is optional DID, it should be nillable imo

Reply:
yes, I agree, bit out of scope for this PR imho